packages/docusaurus/lib/docusaurus/Link.js

```diff
@@ -5,20 +5,28 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React from 'react';
+import React, {useContext} from 'react';
+import Perimeter from 'react-perimeter';
 import {NavLink} from 'react-router-dom';
 
+import DocusaurusContext from '@docusaurus/context';
+
+import preload from '../core/preload';
+
 const externalRegex = /^(https?:|\/\/)/;
 
 function Link(props) {
+  const {routes} = useContext(DocusaurusContext);
   const {to, href} = props;
   const targetLink = to || href;
   const isExternal = externalRegex.test(targetLink);
   return !targetLink || isExternal ? (
     // eslint-disable-next-line jsx-a11y/anchor-has-content
     <a {...props} href={targetLink} />
   ) : (
-    <NavLink {...props} to={targetLink} />
+    <Perimeter padding={20} onBreach={() => preload(routes, targetLink)} once>
```

Comment:
Let's create an optional prop called `preloadProximity` with a default value of 20 that will be supplied to  `<Perimeter padding={props.preloadProximity>`. If `props.preloadProximity` is falsy, don't wrap in `<Perimeter/>`

Reply:
sounds good